index.js

```diff
@@ -429,6 +429,8 @@ FSWatcher.prototype.unwatch = function(paths) {
   paths.forEach(function(path) {
     if (this._closers[path]) {
       this._closers[path]();
+      delete this._closers[path];
+      delete this._getWatchedDir(sysPath.dirname(path)).remove(sysPath.basename(path));
```

Comment:
Yes _closers do not harm but I though it would be better to clean that part too.